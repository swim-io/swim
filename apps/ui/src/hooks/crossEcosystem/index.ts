export * from "./useCoinGeckoPricesQuery";
export * from "./useMultipleUserBalances";
export * from "./useTokensByEcosystem";
export * from "./useUserBalances";
export * from "./useUserNativeBalances";
export * from "./useWallets";
