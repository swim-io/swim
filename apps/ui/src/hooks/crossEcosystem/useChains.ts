import shallow from "zustand/shallow.js";

import type { ChainConfig, EcosystemId } from "../../config";
import { selectConfig } from "../../core/selectors";
import { useEnvironment } from "../../core/store";

export const useChains = (
  ecosystemIds: readonly EcosystemId[],
): readonly (ChainConfig | null)[] => {
  const { ecosystems } = useEnvironment(selectConfig, shallow);
  return ecosystemIds.map((ecosystemId) => {
    const ecosystem = ecosystems.find(({ id }) => id === ecosystemId);
    return ecosystem?.chain ?? null;
  });
};

export const useChain = (ecosystemId: EcosystemId): ChainConfig | null => {
  const [chain] = useChains([ecosystemId]);
  return chain;
};
