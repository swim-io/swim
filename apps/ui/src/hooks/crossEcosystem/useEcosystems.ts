import type {
  SolanaChainConfig,
  SolanaEcosystemConfig,
} from "@swim-io/plugin-ecosystem-solana";
import {
  SOLANA_ECOSYSTEM_ID,
  SOLANA_PROTOCOL,
} from "@swim-io/plugin-ecosystem-solana";
import shallow from "zustand/shallow.js";

import type { EcosystemConfigWithSingleChain, EcosystemId } from "../../config";
import { selectConfig } from "../../core/selectors";
import { useEnvironment } from "../../core/store";

export const useEcosystems = (
  ecosystemIds: readonly EcosystemId[],
): readonly (EcosystemConfigWithSingleChain | null)[] => {
  const { ecosystems } = useEnvironment(selectConfig, shallow);
  return ecosystemIds.map(
    (ecosystemId) => ecosystems.find(({ id }) => id === ecosystemId) ?? null,
  );
};

export const useEcosystem = (
  ecosystemId: EcosystemId | null,
): EcosystemConfigWithSingleChain | null => {
  const [ecosystem] = useEcosystems(ecosystemId ? [ecosystemId] : []);
  return ecosystem ?? null;
};

export const useSolanaEcosystem = (): SolanaEcosystemConfig & {
  readonly chain: SolanaChainConfig;
} => {
  const ecosystem = useEcosystem(SOLANA_ECOSYSTEM_ID);
  if (ecosystem === null || ecosystem.protocol !== SOLANA_PROTOCOL) {
    throw new Error("Invalid Solana ecosystem");
  }
  return ecosystem;
};
