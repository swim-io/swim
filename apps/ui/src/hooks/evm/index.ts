export * from "./useErc20BalanceQuery";
export * from "./useErc20BalancesQuery";
export * from "./useEvmHistoryQuery";
export * from "./useEvmUserNativeBalanceQuery";
export * from "./useEvmWallet";
export * from "./useRegisterErc20Token";
