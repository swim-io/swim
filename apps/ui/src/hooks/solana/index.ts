export * from "./useAccountNftsQuery";
export * from "./useAnchorProvider";
export * from "./useCreateSplTokenAccountsMutation";
export * from "./useSolanaLiquidityQuery";
export * from "./useSolanaConnection";
export * from "./useSolanaWallet";
export * from "./useSolBalanceQuery";
export * from "./useSplTokenAccountsQuery";
export * from "./useSplUserBalance";
