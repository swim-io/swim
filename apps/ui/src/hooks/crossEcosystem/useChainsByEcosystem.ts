import { EvmEcosystemId } from "@swim-io/evm";
import { SOLANA_ECOSYSTEM_ID } from "@swim-io/solana";
import type { ReadonlyRecord } from "@swim-io/utils";
import shallow from "zustand/shallow.js";

import type { ChainSpec, EcosystemId, EvmSpec, SolanaSpec } from "../../config";
import { Protocol } from "../../config";
import { selectConfig } from "../../core/selectors";
import { useEnvironment } from "../../core/store";

export interface ChainsByEcosystem
  extends ReadonlyRecord<EcosystemId, ChainSpec | null> {
  readonly [SOLANA_ECOSYSTEM_ID]: SolanaSpec | null;
  readonly [EvmEcosystemId.Ethereum]: EvmSpec | null;
  readonly [EvmEcosystemId.Bnb]: EvmSpec | null;
  readonly [EvmEcosystemId.Avalanche]: EvmSpec | null;
  readonly [EvmEcosystemId.Polygon]: EvmSpec | null;
  readonly [EvmEcosystemId.Aurora]: EvmSpec | null;
  readonly [EvmEcosystemId.Fantom]: EvmSpec | null;
  readonly [EvmEcosystemId.Karura]: EvmSpec | null;
  readonly [EvmEcosystemId.Acala]: EvmSpec | null;
}

export const useChainsByEcosystem = (): ChainsByEcosystem => {
  const { chains } = useEnvironment(selectConfig, shallow);
  const [solana] = chains[Protocol.Solana];
  const [ethereum, bnb, avalanche, polygon, aurora, fantom, karura, acala] = [
    EvmEcosystemId.Ethereum,
    EvmEcosystemId.Bnb,
    EvmEcosystemId.Avalanche,
    EvmEcosystemId.Polygon,
    EvmEcosystemId.Aurora,
    EvmEcosystemId.Fantom,
    EvmEcosystemId.Karura,
    EvmEcosystemId.Acala,
  ].map(
    (ecosystemId) =>
      chains[Protocol.Evm].find((chain) => chain.ecosystem === ecosystemId) ??
      null,
  );

  return {
    [SOLANA_ECOSYSTEM_ID]: solana,
    [EvmEcosystemId.Ethereum]: ethereum,
    [EvmEcosystemId.Bnb]: bnb,
    [EvmEcosystemId.Avalanche]: avalanche,
    [EvmEcosystemId.Polygon]: polygon,
    [EvmEcosystemId.Aurora]: aurora,
    [EvmEcosystemId.Fantom]: fantom,
    [EvmEcosystemId.Karura]: karura,
    [EvmEcosystemId.Acala]: acala,
  };
};
