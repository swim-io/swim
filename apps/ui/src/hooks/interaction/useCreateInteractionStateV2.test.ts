import { Env } from "@swim-io/core";
import { EvmEcosystemId } from "@swim-io/evm";
import { SOLANA_ECOSYSTEM_ID } from "@swim-io/solana";
import { act, renderHook } from "@testing-library/react-hooks";
import Decimal from "decimal.js";

import {
  CONFIGS,
  DEVNET_POOLS,
  DEVNET_POOLS_FOR_RESTRUCTURE,
  DEVNET_SWIMUSD,
  findTokenById,
} from "../../config";
import { selectConfig } from "../../core/selectors";
import { useEnvironment } from "../../core/store";
import { MOCK_TOKEN_ACCOUNTS, MOCK_WALLETS } from "../../fixtures";
import { Amount, InteractionType, generateId } from "../../models";
import { mockOf, renderHookWithAppContext } from "../../testUtils";
import { useWallets } from "../crossEcosystem";
import { useSplTokenAccountsQuery } from "../solana";

import { useCreateInteractionStateV2 } from "./useCreateInteractionStateV2";

const SOLANA_USDC = findTokenById("devnet-solana-usdc", Env.Devnet);
const SOLANA_USDT = findTokenById("devnet-solana-usdt", Env.Devnet);

Object.defineProperty(global.self, "crypto", {
  value: {
    getRandomValues: (arr: string | readonly any[]) => Buffer.alloc(arr.length),
  },
});

jest.mock("../../models", () => ({
  ...jest.requireActual("../../models"),
  generateId: jest.fn(),
}));

jest.mock("../../core/selectors", () => ({
  ...jest.requireActual("../../core/selectors"),
  selectConfig: jest.fn(),
}));

jest.mock("../crossEcosystem", () => ({
  ...jest.requireActual("../crossEcosystem"),
  useWallets: jest.fn(),
}));

jest.mock("../solana", () => ({
  ...jest.requireActual("../solana"),
  useSplTokenAccountsQuery: jest.fn(),
}));

// Make typescript happy with jest
const generateIdMock = mockOf(generateId);
const selectConfigMock = mockOf(selectConfig);
const useSplTokenAccountsQueryMock = mockOf(useSplTokenAccountsQuery);
const useWalletsMock = mockOf(useWallets);

describe("useCreateInteractionStateV2", () => {
  beforeEach(() => {
    const { result: envStore } = renderHook(() => useEnvironment());
    act(() => {
      envStore.current.setCustomIp("127.0.0.1");
      envStore.current.setEnv(Env.Devnet);
    });
    generateIdMock.mockReturnValue("11111111111111111111111111111111");
    useSplTokenAccountsQueryMock.mockReturnValue({ data: MOCK_TOKEN_ACCOUNTS });
    useWalletsMock.mockReturnValue(MOCK_WALLETS);
    selectConfigMock.mockReturnValue({
      ...CONFIGS[Env.Devnet],
      pools: [...DEVNET_POOLS, ...DEVNET_POOLS_FOR_RESTRUCTURE],
    });
    jest.spyOn(Date, "now").mockImplementation(() => 1657544558283);
  });

  it("should create state for Swap from SOLANA USDC to SOLANA USDT", () => {
    const { result } = renderHookWithAppContext(() =>
      useCreateInteractionStateV2(),
    );
    const createInteractionState = result.current;
    const interactionState = createInteractionState({
      type: InteractionType.SwapV2,
      params: {
        fromTokenData: {
          tokenConfig: findTokenById("devnet-solana-usdc-v2", Env.Devnet),
          ecosystemId: SOLANA_ECOSYSTEM_ID,
          value: new Decimal("1000"),
        },
        toTokenData: {
          tokenConfig: findTokenById("devnet-solana-usdt-v2", Env.Devnet),
          ecosystemId: SOLANA_ECOSYSTEM_ID,
          value: new Decimal("1000"),
        },
      },
    });
    expect(interactionState).toMatchSnapshot();
  });

  it("should create state for Swap from ETHEREUM USDC to ETHEREUM USDT", () => {
    const { result } = renderHookWithAppContext(() =>
      useCreateInteractionStateV2(),
    );
    const createInteractionState = result.current;
    const interactionState = createInteractionState({
      type: InteractionType.SwapV2,
      params: {
        fromTokenData: {
          tokenConfig: findTokenById("devnet-ethereum-usdc", Env.Devnet),
          ecosystemId: EvmEcosystemId.Ethereum,
          value: new Decimal("1000"),
        },
        toTokenData: {
          tokenConfig: findTokenById("devnet-ethereum-usdt", Env.Devnet),
          ecosystemId: EvmEcosystemId.Ethereum,
          value: new Decimal("1000"),
        },
      },
    });
    expect(interactionState).toMatchSnapshot();
  });

  it("should create state for Swap from SOLANA USDC to ETHEREUM USDC", () => {
    const { result } = renderHookWithAppContext(() =>
      useCreateInteractionStateV2(),
    );
    const createInteractionState = result.current;
    const interactionState = createInteractionState({
      type: InteractionType.SwapV2,
      params: {
        fromTokenData: {
          tokenConfig: findTokenById("devnet-solana-usdc-v2", Env.Devnet),
          ecosystemId: SOLANA_ECOSYSTEM_ID,
          value: new Decimal("1000"),
        },
        toTokenData: {
          tokenConfig: findTokenById("devnet-ethereum-usdc", Env.Devnet),
          ecosystemId: EvmEcosystemId.Ethereum,
          value: new Decimal("1000"),
        },
      },
    });
    expect(interactionState).toMatchSnapshot();
  });

  it("should create state for Swap from ETHEREUM USDC to SOLANA USDC", () => {
    const { result } = renderHookWithAppContext(() =>
      useCreateInteractionStateV2(),
    );
    const createInteractionState = result.current;
    const interactionState = createInteractionState({
      type: InteractionType.SwapV2,
      params: {
        fromTokenData: {
          tokenConfig: findTokenById("devnet-ethereum-usdc", Env.Devnet),
          ecosystemId: EvmEcosystemId.Ethereum,
          value: new Decimal("1000"),
        },
        toTokenData: {
          tokenConfig: findTokenById("devnet-solana-usdc-v2", Env.Devnet),
          ecosystemId: SOLANA_ECOSYSTEM_ID,
          value: new Decimal("1000"),
        },
      },
    });
    expect(interactionState).toMatchSnapshot();
  });

  it("should create state for Swap from ETHEREUM USDC to BNB USDT", () => {
    const { result } = renderHookWithAppContext(() =>
      useCreateInteractionStateV2(),
    );
    const createInteractionState = result.current;
    const interactionState = createInteractionState({
      type: InteractionType.SwapV2,
      params: {
        fromTokenData: {
          tokenConfig: findTokenById("devnet-ethereum-usdc", Env.Devnet),
          ecosystemId: EvmEcosystemId.Ethereum,
          value: new Decimal("1000"),
        },
        toTokenData: {
          tokenConfig: findTokenById("devnet-bnb-usdt", Env.Devnet),
          ecosystemId: EvmEcosystemId.Bnb,
          value: new Decimal("1000"),
        },
      },
    });
    expect(interactionState).toMatchSnapshot();
  });

  it("should create state for Add", () => {
    const { result } = renderHookWithAppContext(() =>
      useCreateInteractionStateV2(),
    );
    const createInteractionState = result.current;
    const interactionState = createInteractionState({
      type: InteractionType.Add,
      poolId: "devnet-solana-usdc-usdt",
      params: {
        inputAmounts: [
          Amount.fromHuman(SOLANA_USDC, new Decimal("100")),
          Amount.fromHuman(SOLANA_USDT, new Decimal("10")),
        ],
        minimumMintAmount: Amount.fromHuman(DEVNET_SWIMUSD, new Decimal("110")),
      },
      lpTokenTargetEcosystem: SOLANA_ECOSYSTEM_ID,
    });
    expect(interactionState).toMatchSnapshot();
  });

  it("should create state for RemoveExactOutput", () => {
    const { result } = renderHookWithAppContext(() =>
      useCreateInteractionStateV2(),
    );
    const createInteractionState = result.current;
    const interactionState = createInteractionState({
      type: InteractionType.RemoveExactOutput,
      poolId: "devnet-solana-usdc-usdt",
      params: {
        exactOutputAmounts: [
          Amount.fromHuman(SOLANA_USDC, new Decimal("100")),
          Amount.fromHuman(SOLANA_USDT, new Decimal("10")),
        ],
        maximumBurnAmount: Amount.fromHuman(DEVNET_SWIMUSD, new Decimal("110")),
      },
      lpTokenSourceEcosystem: SOLANA_ECOSYSTEM_ID,
    });
    expect(interactionState).toMatchSnapshot();
  });

  it("should create state for RemoveExactBurn", () => {
    const { result } = renderHookWithAppContext(() =>
      useCreateInteractionStateV2(),
    );
    const createInteractionState = result.current;
    const interactionState = createInteractionState({
      type: InteractionType.RemoveExactBurn,
      poolId: "devnet-solana-usdc-usdt",
      params: {
        minimumOutputAmount: Amount.fromHuman(SOLANA_USDC, new Decimal("110")),
        exactBurnAmount: Amount.fromHuman(DEVNET_SWIMUSD, new Decimal("110")),
      },
      lpTokenSourceEcosystem: SOLANA_ECOSYSTEM_ID,
    });
    expect(interactionState).toMatchSnapshot();
  });

  it("should create state for RemoveUniform", () => {
    const { result } = renderHookWithAppContext(() =>
      useCreateInteractionStateV2(),
    );
    const createInteractionState = result.current;
    const interactionState = createInteractionState({
      type: InteractionType.RemoveUniform,
      poolId: "devnet-solana-usdc-usdt",
      params: {
        minimumOutputAmounts: [
          Amount.fromHuman(SOLANA_USDC, new Decimal("100")),
          Amount.fromHuman(SOLANA_USDT, new Decimal("10")),
        ],
        exactBurnAmount: Amount.fromHuman(DEVNET_SWIMUSD, new Decimal("110")),
      },
      lpTokenSourceEcosystem: SOLANA_ECOSYSTEM_ID,
    });
    expect(interactionState).toMatchSnapshot();
  });
});
