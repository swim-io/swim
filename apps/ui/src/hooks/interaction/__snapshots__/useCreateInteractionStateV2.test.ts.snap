// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useCreateInteractionStateV2 should create state for Add 1`] = `
Object {
  "addTxId": null,
  "approvalTxIds": Array [],
  "interaction": Object {
    "connectedWallets": Object {
      "acala": null,
      "aptos": null,
      "aurora": null,
      "avalanche": null,
      "bnb": null,
      "ethereum": null,
      "fantom": null,
      "karura": null,
      "polygon": null,
      "solana": "6sbzC1eH4FTujJXWj51eQe25cYvr4xfXbJ1vAj7j2k5J",
    },
    "env": "Testnet",
    "id": "11111111111111111111111111111111",
    "lpTokenTargetEcosystem": "solana",
    "params": Object {
      "inputAmounts": Array [
        "100",
        "10",
      ],
      "minimumMintAmount": "110",
    },
    "poolId": "testnet-solana-usdc-usdt",
    "poolIds": Array [
      "testnet-solana-usdc-usdt",
    ],
    "submittedAt": 1657544558283,
    "type": 1,
  },
  "interactionType": 1,
  "requiredSplTokenAccounts": Object {
    "2w7wsGofEAvLiWXZgJySXZ4gofEhm8jQ9rtwXr1zbzUc": Object {
      "isExistingAccount": false,
      "txId": null,
    },
    "3ngTtoyP9GFybFifX1dr7gCFXFiM2Wr6NfXn6EuU7k6C": Object {
      "isExistingAccount": false,
      "txId": null,
    },
    "DznJzVAjPHBvyyqXEQgPWTonF2nhwoSoutPNbXjmsUvY": Object {
      "isExistingAccount": false,
      "txId": null,
    },
  },
  "version": 2,
}
`;

exports[`useCreateInteractionStateV2 should create state for RemoveExactBurn 1`] = `
Object {
  "approvalTxIds": Array [],
  "interaction": Object {
    "connectedWallets": Object {
      "acala": null,
      "aptos": null,
      "aurora": null,
      "avalanche": null,
      "bnb": null,
      "ethereum": null,
      "fantom": null,
      "karura": null,
      "polygon": null,
      "solana": "6sbzC1eH4FTujJXWj51eQe25cYvr4xfXbJ1vAj7j2k5J",
    },
    "env": "Testnet",
    "id": "11111111111111111111111111111111",
    "lpTokenSourceEcosystem": "solana",
    "params": Object {
      "exactBurnAmount": "110",
      "minimumOutputAmount": "110",
    },
    "poolId": "testnet-solana-usdc-usdt",
    "poolIds": Array [
      "testnet-solana-usdc-usdt",
    ],
    "submittedAt": 1657544558283,
    "type": 3,
  },
  "interactionType": 3,
  "removeTxId": null,
  "requiredSplTokenAccounts": Object {
    "2w7wsGofEAvLiWXZgJySXZ4gofEhm8jQ9rtwXr1zbzUc": Object {
      "isExistingAccount": false,
      "txId": null,
    },
    "3ngTtoyP9GFybFifX1dr7gCFXFiM2Wr6NfXn6EuU7k6C": Object {
      "isExistingAccount": false,
      "txId": null,
    },
  },
  "version": 2,
}
`;

exports[`useCreateInteractionStateV2 should create state for RemoveExactOutput 1`] = `
Object {
  "approvalTxIds": Array [],
  "interaction": Object {
    "connectedWallets": Object {
      "acala": null,
      "aptos": null,
      "aurora": null,
      "avalanche": null,
      "bnb": null,
      "ethereum": null,
      "fantom": null,
      "karura": null,
      "polygon": null,
      "solana": "6sbzC1eH4FTujJXWj51eQe25cYvr4xfXbJ1vAj7j2k5J",
    },
    "env": "Testnet",
    "id": "11111111111111111111111111111111",
    "lpTokenSourceEcosystem": "solana",
    "params": Object {
      "exactOutputAmounts": Array [
        "100",
        "10",
      ],
      "maximumBurnAmount": "110",
    },
    "poolId": "testnet-solana-usdc-usdt",
    "poolIds": Array [
      "testnet-solana-usdc-usdt",
    ],
    "submittedAt": 1657544558283,
    "type": 4,
  },
  "interactionType": 4,
  "removeTxId": null,
  "requiredSplTokenAccounts": Object {
    "2w7wsGofEAvLiWXZgJySXZ4gofEhm8jQ9rtwXr1zbzUc": Object {
      "isExistingAccount": false,
      "txId": null,
    },
    "3ngTtoyP9GFybFifX1dr7gCFXFiM2Wr6NfXn6EuU7k6C": Object {
      "isExistingAccount": false,
      "txId": null,
    },
    "DznJzVAjPHBvyyqXEQgPWTonF2nhwoSoutPNbXjmsUvY": Object {
      "isExistingAccount": false,
      "txId": null,
    },
  },
  "version": 2,
}
`;

exports[`useCreateInteractionStateV2 should create state for RemoveUniform 1`] = `
Object {
  "approvalTxIds": Array [],
  "interaction": Object {
    "connectedWallets": Object {
      "acala": null,
      "aptos": null,
      "aurora": null,
      "avalanche": null,
      "bnb": null,
      "ethereum": null,
      "fantom": null,
      "karura": null,
      "polygon": null,
      "solana": "6sbzC1eH4FTujJXWj51eQe25cYvr4xfXbJ1vAj7j2k5J",
    },
    "env": "Testnet",
    "id": "11111111111111111111111111111111",
    "lpTokenSourceEcosystem": "solana",
    "params": Object {
      "exactBurnAmount": "110",
      "minimumOutputAmounts": Array [
        "100",
        "10",
      ],
    },
    "poolId": "testnet-solana-usdc-usdt",
    "poolIds": Array [
      "testnet-solana-usdc-usdt",
    ],
    "submittedAt": 1657544558283,
    "type": 2,
  },
  "interactionType": 2,
  "removeTxId": null,
  "requiredSplTokenAccounts": Object {
    "2w7wsGofEAvLiWXZgJySXZ4gofEhm8jQ9rtwXr1zbzUc": Object {
      "isExistingAccount": false,
      "txId": null,
    },
    "3ngTtoyP9GFybFifX1dr7gCFXFiM2Wr6NfXn6EuU7k6C": Object {
      "isExistingAccount": false,
      "txId": null,
    },
    "DznJzVAjPHBvyyqXEQgPWTonF2nhwoSoutPNbXjmsUvY": Object {
      "isExistingAccount": false,
      "txId": null,
    },
  },
  "version": 2,
}
`;

exports[`useCreateInteractionStateV2 should create state for Swap from ETHEREUM USDC to BNB USDT 1`] = `
Object {
  "approvalTxIds": Array [],
  "interaction": Object {
    "connectedWallets": Object {
      "acala": null,
      "aptos": null,
      "aurora": null,
      "avalanche": null,
      "bnb": "0x90f8bf6a479f320ead074411a4b0e7944ea8c9c1",
      "ethereum": "0x90f8bf6a479f320ead074411a4b0e7944ea8c9c1",
      "fantom": null,
      "karura": null,
      "polygon": null,
      "solana": null,
    },
    "env": "Testnet",
    "id": "11111111111111111111111111111111",
    "params": Object {
      "fromTokenData": Object {
        "ecosystemId": "ethereum",
        "tokenConfig": Object {
          "id": "testnet-ethereum-usdc",
          "nativeDetails": Object {
            "address": "0x45B167CF5b14007Ca0490dCfB7C4B870Ec0C0Aa6",
            "decimals": 6,
          },
          "nativeEcosystemId": "ethereum",
          "projectId": "usdc",
          "wrappedDetails": Map {
            "solana" => Object {
              "address": "7Lf95y8NuCU5RRC95oUtbBtckPAtbr9ubTgrCiyZ1kEf",
              "decimals": 6,
            },
          },
        },
        "value": "1000",
      },
      "toTokenData": Object {
        "ecosystemId": "bnb",
        "tokenConfig": Object {
          "id": "testnet-bnb-usdt",
          "nativeDetails": Object {
            "address": "0x98529E942FD121d9C470c3d4431A008257E0E714",
            "decimals": 18,
          },
          "nativeEcosystemId": "bnb",
          "projectId": "usdt",
          "wrappedDetails": Map {
            "solana" => Object {
              "address": "4dr6ogcLsaFf2RDF4LJU1CvNtNKxonVqQvM6vuGdVR1e",
              "decimals": 8,
            },
          },
        },
        "value": "1000",
      },
    },
    "poolIds": Array [
      "testnet-ethereum-usdc-usdt",
      "testnet-bnb-busd-usdt",
    ],
    "submittedAt": 1657544558283,
    "type": 5,
  },
  "interactionType": 5,
  "receiveAndSwapTxId": null,
  "swapAndTransferTxId": null,
  "swapType": "CrossChainEvmToEvm",
  "version": 2,
}
`;

exports[`useCreateInteractionStateV2 should create state for Swap from ETHEREUM USDC to ETHEREUM USDT 1`] = `
Object {
  "approvalTxIds": Array [],
  "interaction": Object {
    "connectedWallets": Object {
      "acala": null,
      "aptos": null,
      "aurora": null,
      "avalanche": null,
      "bnb": null,
      "ethereum": "0x90f8bf6a479f320ead074411a4b0e7944ea8c9c1",
      "fantom": null,
      "karura": null,
      "polygon": null,
      "solana": null,
    },
    "env": "Testnet",
    "id": "11111111111111111111111111111111",
    "params": Object {
      "fromTokenData": Object {
        "ecosystemId": "ethereum",
        "tokenConfig": Object {
          "id": "testnet-ethereum-usdc",
          "nativeDetails": Object {
            "address": "0x45B167CF5b14007Ca0490dCfB7C4B870Ec0C0Aa6",
            "decimals": 6,
          },
          "nativeEcosystemId": "ethereum",
          "projectId": "usdc",
          "wrappedDetails": Map {
            "solana" => Object {
              "address": "7Lf95y8NuCU5RRC95oUtbBtckPAtbr9ubTgrCiyZ1kEf",
              "decimals": 6,
            },
          },
        },
        "value": "1000",
      },
      "toTokenData": Object {
        "ecosystemId": "ethereum",
        "tokenConfig": Object {
          "id": "testnet-ethereum-usdt",
          "nativeDetails": Object {
            "address": "0x996f42BdB0CB71F831C2eFB05Ac6d0d226979e5B",
            "decimals": 6,
          },
          "nativeEcosystemId": "ethereum",
          "projectId": "usdt",
          "wrappedDetails": Map {
            "solana" => Object {
              "address": "8Cfyi1mYXqKATUkMPvb9BMXikdbppJst6E7eQJkKjAtf",
              "decimals": 6,
            },
          },
        },
        "value": "1000",
      },
    },
    "poolIds": Array [
      "testnet-ethereum-usdc-usdt",
    ],
    "submittedAt": 1657544558283,
    "type": 5,
  },
  "interactionType": 5,
  "onChainSwapTxId": null,
  "swapType": "SingleChainEvm",
  "version": 2,
}
`;

exports[`useCreateInteractionStateV2 should create state for Swap from ETHEREUM USDC to SOLANA USDC 1`] = `
Object {
  "approvalTxIds": Array [],
  "claimTokenOnSolanaTxId": null,
  "interaction": Object {
    "connectedWallets": Object {
      "acala": null,
      "aptos": null,
      "aurora": null,
      "avalanche": null,
      "bnb": null,
      "ethereum": "0x90f8bf6a479f320ead074411a4b0e7944ea8c9c1",
      "fantom": null,
      "karura": null,
      "polygon": null,
      "solana": "6sbzC1eH4FTujJXWj51eQe25cYvr4xfXbJ1vAj7j2k5J",
    },
    "env": "Testnet",
    "id": "11111111111111111111111111111111",
    "params": Object {
      "fromTokenData": Object {
        "ecosystemId": "ethereum",
        "tokenConfig": Object {
          "id": "testnet-ethereum-usdc",
          "nativeDetails": Object {
            "address": "0x45B167CF5b14007Ca0490dCfB7C4B870Ec0C0Aa6",
            "decimals": 6,
          },
          "nativeEcosystemId": "ethereum",
          "projectId": "usdc",
          "wrappedDetails": Map {
            "solana" => Object {
              "address": "7Lf95y8NuCU5RRC95oUtbBtckPAtbr9ubTgrCiyZ1kEf",
              "decimals": 6,
            },
          },
        },
        "value": "1000",
      },
      "toTokenData": Object {
        "ecosystemId": "solana",
        "tokenConfig": Object {
          "id": "testnet-solana-usdc-v2",
          "nativeDetails": Object {
            "address": "6iSRgpK4oiqJZuhpLsTecW3n9xBKUq9N3VPQN7RinYwq",
            "decimals": 6,
          },
          "nativeEcosystemId": "solana",
          "projectId": "usdc",
          "wrappedDetails": Map {},
        },
        "value": "1000",
      },
    },
    "poolIds": Array [
      "testnet-ethereum-usdc-usdt",
      "testnet-solana-usdc-usdt",
    ],
    "submittedAt": 1657544558283,
    "type": 5,
  },
  "interactionType": 5,
  "postVaaOnSolanaTxIds": Array [],
  "requiredSplTokenAccounts": Object {
    "3ngTtoyP9GFybFifX1dr7gCFXFiM2Wr6NfXn6EuU7k6C": Object {
      "isExistingAccount": false,
      "txId": null,
    },
    "6iSRgpK4oiqJZuhpLsTecW3n9xBKUq9N3VPQN7RinYwq": Object {
      "isExistingAccount": false,
      "txId": null,
    },
  },
  "signatureSetAddress": null,
  "swapAndTransferTxId": null,
  "swapFromSwimUsdTxId": null,
  "swapType": "CrossChainEvmToSolana",
  "version": 2,
}
`;

exports[`useCreateInteractionStateV2 should create state for Swap from SOLANA USDC to ETHEREUM USDC 1`] = `
Object {
  "interaction": Object {
    "connectedWallets": Object {
      "acala": null,
      "aptos": null,
      "aurora": null,
      "avalanche": null,
      "bnb": null,
      "ethereum": "0x90f8bf6a479f320ead074411a4b0e7944ea8c9c1",
      "fantom": null,
      "karura": null,
      "polygon": null,
      "solana": "6sbzC1eH4FTujJXWj51eQe25cYvr4xfXbJ1vAj7j2k5J",
    },
    "env": "Testnet",
    "id": "11111111111111111111111111111111",
    "params": Object {
      "fromTokenData": Object {
        "ecosystemId": "solana",
        "tokenConfig": Object {
          "id": "testnet-solana-usdc-v2",
          "nativeDetails": Object {
            "address": "6iSRgpK4oiqJZuhpLsTecW3n9xBKUq9N3VPQN7RinYwq",
            "decimals": 6,
          },
          "nativeEcosystemId": "solana",
          "projectId": "usdc",
          "wrappedDetails": Map {},
        },
        "value": "1000",
      },
      "toTokenData": Object {
        "ecosystemId": "ethereum",
        "tokenConfig": Object {
          "id": "testnet-ethereum-usdc",
          "nativeDetails": Object {
            "address": "0x45B167CF5b14007Ca0490dCfB7C4B870Ec0C0Aa6",
            "decimals": 6,
          },
          "nativeEcosystemId": "ethereum",
          "projectId": "usdc",
          "wrappedDetails": Map {
            "solana" => Object {
              "address": "7Lf95y8NuCU5RRC95oUtbBtckPAtbr9ubTgrCiyZ1kEf",
              "decimals": 6,
            },
          },
        },
        "value": "1000",
      },
    },
    "poolIds": Array [
      "testnet-solana-usdc-usdt",
      "testnet-ethereum-usdc-usdt",
    ],
    "submittedAt": 1657544558283,
    "type": 5,
  },
  "interactionType": 5,
  "receiveAndSwapTxId": null,
  "requiredSplTokenAccounts": Object {
    "3ngTtoyP9GFybFifX1dr7gCFXFiM2Wr6NfXn6EuU7k6C": Object {
      "isExistingAccount": false,
      "txId": null,
    },
    "6iSRgpK4oiqJZuhpLsTecW3n9xBKUq9N3VPQN7RinYwq": Object {
      "isExistingAccount": false,
      "txId": null,
    },
  },
  "swapToSwimUsdTxId": null,
  "swapType": "CrossChainSolanaToEvm",
  "transferSwimUsdToEvmTxId": null,
  "version": 2,
}
`;

exports[`useCreateInteractionStateV2 should create state for Swap from SOLANA USDC to SOLANA USDT 1`] = `
Object {
  "interaction": Object {
    "connectedWallets": Object {
      "acala": null,
      "aptos": null,
      "aurora": null,
      "avalanche": null,
      "bnb": null,
      "ethereum": null,
      "fantom": null,
      "karura": null,
      "polygon": null,
      "solana": "6sbzC1eH4FTujJXWj51eQe25cYvr4xfXbJ1vAj7j2k5J",
    },
    "env": "Testnet",
    "id": "11111111111111111111111111111111",
    "params": Object {
      "fromTokenData": Object {
        "ecosystemId": "solana",
        "tokenConfig": Object {
          "id": "testnet-solana-usdc-v2",
          "nativeDetails": Object {
            "address": "6iSRgpK4oiqJZuhpLsTecW3n9xBKUq9N3VPQN7RinYwq",
            "decimals": 6,
          },
          "nativeEcosystemId": "solana",
          "projectId": "usdc",
          "wrappedDetails": Map {},
        },
        "value": "1000",
      },
      "toTokenData": Object {
        "ecosystemId": "solana",
        "tokenConfig": Object {
          "id": "testnet-solana-usdt-v2",
          "nativeDetails": Object {
            "address": "8VbikoRxEoyYzTDzDcPTSsGk2E5mM7fK1WrVpKrVd75M",
            "decimals": 6,
          },
          "nativeEcosystemId": "solana",
          "projectId": "usdt",
          "wrappedDetails": Map {},
        },
        "value": "1000",
      },
    },
    "poolIds": Array [
      "testnet-solana-usdc-usdt",
    ],
    "submittedAt": 1657544558283,
    "type": 5,
  },
  "interactionType": 5,
  "onChainSwapTxId": null,
  "requiredSplTokenAccounts": Object {
    "6iSRgpK4oiqJZuhpLsTecW3n9xBKUq9N3VPQN7RinYwq": Object {
      "isExistingAccount": false,
      "txId": null,
    },
    "8VbikoRxEoyYzTDzDcPTSsGk2E5mM7fK1WrVpKrVd75M": Object {
      "isExistingAccount": false,
      "txId": null,
    },
  },
  "swapType": "SingleChainSolana",
  "version": 2,
}
`;
